### ISSUE TITLE
Rewrite using native webview components

### ISSUE THREAD

**Issue by prologic** [prologic/gopherclient#4]
See https://github.com/prologic/gopherclient/tree/webview for WIP

This supercedes #3 

QML/QT turns out to be hard to deploy across other platforms such as Linux/Windows.

**Comment by k4k** [comment]
Tried to build from webview branch and receiving the following error. I'm running Fedora 26. Do you know what webkit packages I need?

```
# pkg-config --cflags gtk+-3.0 webkitgtk-3.0
Package webkitgtk-3.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `webkitgtk-3.0.pc'
to the PKG_CONFIG_PATH environment variable
Package 'webkitgtk-3.0', required by 'virtual:world', not found
pkg-config: exit status 1
```

**Comment by k4k** [comment]
Follow up to previous comment. I installed `webkitgtk3-devel` and now build fails as follows:
```
# github.com/prologic/gopherclient
./main.go:68: too many arguments to conversion to gopherproxy.Handler: gopherproxy.Handler(tpl, uri)
```

**Comment by prologic** [comment]
That's bizarre; I don't get a compile error on my OS X laptop. :/ I don't have a Linux dev environment I can easily test with except for Docker-based ones.

**Comment by k4k** [comment]
Can you give me some additional information about your environment? Mostly
interested in what version of Go are you using and what your go environment
variables look like. I have OSX at work and can try to see what it takes to
replicate the error there when I get in the office later today.

**Comment by k4k** [comment]
OK, I'm now on my Mac and have tested, and replicated the issue. Here are the replication steps. See if you can reproduce the issue on your end following the steps below. I suspect that this is an issue with a newer version of a library (probably and update to `net/http` specific to `http.HanderFunc()` ).

```#!bash
$ git clone git@github.com:prologic/gopherclient.git
$ git co -b webview origin/webview
$ go build
```

This should produce an error similar to this:

```
main.go:15:2: cannot find package "github.com/mitchellh/go-homedir" in any of:
        /Users/twood/go/src/github.com/prologic/gopherclient/vendor/github.com/mitchellh/go-homedir (vendor tree)
        /usr/local/Cellar/go/1.9.1/libexec/src/github.com/mitchellh/go-homedir (from $GOROOT)
        /Users/twood/go/src/github.com/mitchellh/go-homedir (from $GOPATH)
main.go:16:2: cannot find package "github.com/prologic/go-gopher" in any of:
        /Users/twood/go/src/github.com/prologic/gopherclient/vendor/github.com/prologic/go-gopher (vendor tree)
        /usr/local/Cellar/go/1.9.1/libexec/src/github.com/prologic/go-gopher (from $GOROOT)
        /Users/twood/go/src/github.com/prologic/go-gopher (from $GOPATH)
main.go:17:2: cannot find package "github.com/prologic/gopherproxy" in any of:
        /Users/twood/go/src/github.com/prologic/gopherclient/vendor/github.com/prologic/gopherproxy (vendor tree)
        /usr/local/Cellar/go/1.9.1/libexec/src/github.com/prologic/gopherproxy (from $GOROOT)
        /Users/twood/go/src/github.com/prologic/gopherproxy (from $GOPATH)
main.go:18:2: cannot find package "github.com/zserge/webview" in any of:
        /Users/twood/go/src/github.com/prologic/gopherclient/vendor/github.com/zserge/webview (vendor tree)
        /usr/local/Cellar/go/1.9.1/libexec/src/github.com/zserge/webview (from $GOROOT)
        /Users/twood/go/src/github.com/zserge/webview (from $GOPATH)
```

If it does not, simulate not having anything in your $GOPATH/src directory by running the following (we want to try this with the latest dependency versions since that's what anyone building the project right now has to do because `vendor/` is missing in the webview):

```#!bash
$ mkdir src bin pkg
$ export GOPATH=./
```

Then Try to build again:

```#!bash
$ go build
```

Once you've successfully received errors about missing dependencies, fetch them using whatever means you are familiar with, be it using `dep init` or `vendetta`. This will fetch the latest copies of all libraries. Try to build again:

```#bash
$ go build
```

This should produce an error like:

```#!bash
# github.com/prologic/gopherclient
./main.go:68: too many arguments to conversion to gopherproxy.Handler: gopherproxy.Handler(tpl, uri)
```

I'm going to continue to dig in to the versions you have fetched with `vendetta` to see if I can isolate this issue further.

**Comment by k4k** [comment]
So, this is very interesting. I was able to successfully get a build working by using `vendetta` to initialize the dependencies in `.gitmodules`. I tested this procedure on both OSX and Fedora (within docker using fedora:latest) and it builds fine! Now I'm really confused as to what's different between vendoring your dependencies using `dep init` and `vendetta`. From looking at the `.gitignore` file and my limited understanding of `vendetta` there should be no difference. I'll look in to this further and get back to you :)

**Comment by prologic** [comment]
I did some more R&D into dependency management in Go (*I did before too*) and I personally find the git submodule + `vendor/` approach the simplest and most effective. The tool I have found to work the best to achieve this is `vendetta`. I'm not convinced Godep is really all that great; but it looks like it was the first "kid on the block" so to speak and supports Go 1+ and 1.4+ (*`vendor/` was only introduced in 1.5+*).

Let me commit vendored packages.
